### PR TITLE
Migrate `prefect-azure` to `prefect` repository

### PR DESCRIPTION
**Pull request by desertaxle**
- Creates src/integrations directory for integrations packages
- Migrates code for `prefect-azure` to the `prefect` repository and adds a workflow for running tests for integrations packages.
- Updates `prefect-azure` to use a `pyproject.toml` file and `setuptools-scm` for version discovery via git tags instead of `versioneer`

## Next steps
- Add release workflow for integrations packages
- Add to docs tree

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: .github/workflows/integration-pacakge-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        id: set-matrix
+        run: |
+          if [[ $GITHUB_EVENT_NAME == 'pull_request' ]]; then
+            COMMIT_RANGE="${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}"
+          else
+            COMMIT_RANGE="${{ github.event.before }}..${{ github.event.after }}"
+          fi
+          python scripts/generate_integration_package_tests_matrix.py "$COMMIT_RANGE" > matrix.json
+          cat matrix.json
+          echo "matrix=$(cat matrix.json)" >> $GITHUB_OUTPUT
```
Comment (desertaxle) [inline .github/workflows/integration-pacakge-tests.yaml:37]: This `prepare-matrix` job ensures that we only run tests for integrations packages with changes.

File: scripts/generate_integration_package_tests_matrix.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            matrix["include"].append({"package": package, "python-version": version})
+    return matrix
+
+
+if __name__ == "__main__":
+    # Get the range of commits to check, or use a default range
+    commit_range = sys.argv[1] if len(sys.argv) > 1 else "HEAD~1..HEAD"
+    packages = get_changed_packages(commit_range=commit_range)
+    matrix = generate_matrix(packages, PYTHON_VERSIONS)
+    print(json.dumps(matrix))
```
Comment (desertaxle) [inline scripts/generate_integration_package_tests_matrix.py:47]: This script gets integrations packages with changes from the git history

File: src/integrations/prefect-azure/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Homepage = "https://github.com/PrefectHQ/prefect-azure"
+
+[project.entry-points."prefect.collections"]
+prefect_azure = "prefect_azure"
+
+[tool.setuptools_scm]
+version_file = "prefect_azure/_version.py"
+root = "../../.."
+tag_regex = "^prefect-azure-(?P<version>\\d+\\.\\d+\\.\\d+)$"
+fallback_version = "0.0.0"
```
Comment (desertaxle) [inline src/integrations/prefect-azure/pyproject.toml:79]: These settings ensure that only tags prefixed with `prefect-azure-` will be used to determine the `prefect-azure` version.

**Review by chrisguidry — APPROVED** [review]
I dig it!

File: scripts/generate_integration_package_tests_matrix.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    matrix = {"include": []}
+    for package in packages:
+        for version in python_versions:
+            matrix["include"].append({"package": package, "python-version": version})
+    return matrix
+
+
+if __name__ == "__main__":
+    # Get the range of commits to check, or use a default range
+    commit_range = sys.argv[1] if len(sys.argv) > 1 else "HEAD~1..HEAD"
```
Comment (zzstoatzz) [inline scripts/generate_integration_package_tests_matrix.py:43]: ahhh nice!

**Review by zzstoatzz — APPROVED** [review]
🐎

**Pull request merged**